pkg/controllers/cidrs_controller.go

```diff
@@ -103,25 +106,130 @@ func (r *CIDRReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	return ctrl.Result{}, nil
 }
 
+// trimSpaces removes leading and trailing spaces from each string in the slice
+func trimSpaces(values []string) []string {
+	result := make([]string, 0, len(values))
+	for _, v := range values {
+		trimmed := strings.TrimSpace(v)
+		if trimmed != "" {
+			result = append(result, trimmed)
+		}
+	}
+	return result
+}
+
 func applyProcessor(reader io.Reader, processing ipamv1alpha1.Processing) ([]string, error) {
-	if processing.JSONPath != "" {
+	// Default to YAML format if not specified (backward compatibility)
```

Comment:
I would advocate for splitting this in various processors per type and leave just a switch case in the main one

```
func applyProcessor(reader io.Reader, processing ipamv1alpha1.Processing) ([]string, error) {
    switch processing.Format {
        case ipamv1alpha1.CommaSeparatedValues:
            return processCSV(reader, processing)
        case ipamv1alpha1.LineSeparatedValues:
            return processLSV(reader, processing)
        case ipamv1alpha1.CombinedSeparatedValues:
            return processCombinedSeparatedValues(reader, processing)
        default:
            return processYAML(reader, processor)
    }
}
```